### DISCUSSION TITLE
Is it possible to render the opcode also as an operand? Understanding the error "Constraining currently undefined operand"

### DISCUSSION THREAD

**Discussion by hippietrail** [NationalSecurityAgency/ghidra#7011] [Q&A]
As a kind of `printf`-style debugging while working out my language spec I've tried to render an instruction with its opcode as if it's also an operand.

I always get the error "Constraining currently undefined operand", which doesn't help me.

Is this just something that's never possible and happens to trigger an error message intended for something else? Or is it possible and I'm just not doing it the right way?

Things I've tried:
```
:HTEX op is op=0x0 { }
```
```
:HTEX op is op=0x0 & op { }
```
```
:HTEX op is op=0x0 ;op { }
```
I got here because I'm progressing from opcodes based on all 8 bits in a byte to ones that use some bits as the opcode and some as built-in operand and trying to output what I've got so far to make sure I'm on track. When I couldn't get it to work I tried with whole opcodes which I actually thought I had done yesterday. Now I'm just thoroughly lost.

**Comment by konchunas** [comment]
For me `:HTEX op is op=0x0 & op { }` sequence compiles without errors.
Could this be an error in another place in your architecture? Maybe somewhere in token definitions?

**Reply by GhidorahRex** [reply]
I usually just duplicate the token definition:

```
define token optoken (8)
op=(5,7)
opid=(5,7)
;

HTEX opid is op=0x0 & opid {}
```
